src/octokit.ts

```diff
@@ -2,43 +2,91 @@ import {getInput, warning} from '@actions/core'
 import {Octokit} from '@octokit/action'
 import {retry} from '@octokit/plugin-retry'
 import {throttling} from '@octokit/plugin-throttling'
+import dotenv from 'dotenv'
+import {createAppAuth} from '@octokit/auth-app'
+// // eslint-disable-next-line import/no-commonjs
+// import { Octokit } from '@octokit/rest';
+
+dotenv.config()
 
 const token = getInput('token') || process.env.GITHUB_TOKEN
 
 const RetryAndThrottlingOctokit = Octokit.plugin(throttling, retry)
 
-export const octokit = new RetryAndThrottlingOctokit({
-  auth: `token ${token}`,
-  throttle: {
-    onRateLimit: (
-      retryAfter: number,
-      options: any,
-      _o: any,
-      retryCount: number
-    ) => {
-      warning(
-        `Request quota exhausted for request ${options.method} ${options.url}
+console.log(process.env.NODE_DEV)
+
+export const octokit = process.env.NODE_DEV
```

Comment:
These changes are only required when running outside the github actions runner. Basically, it authenticates with github - but when running as a github action, the authentication is taken care of by the runner config.